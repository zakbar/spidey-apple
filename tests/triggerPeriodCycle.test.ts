import {
  INetworkProvider,
  Orchard,
  Town,
  triggerPeriodCycle,
} from "../src/network";

const townA = new Town("A", 100);
const townB = new Town("B", 90);
const townC = new Town("C", 80);
const townD = new Town("D", 0);
const orchard1 = new Orchard("1", 4, true, 20, [townA, townB]);
const orchard2 = new Orchard("2", 3, true, 15, [townA, townC]);
const orchard3 = new Orchard("3", 2, true, 7, [townC, townD]);

const mockNetworkProvider: INetworkProvider = {
  getNetworks: () => {
    return [
      {
        getSize: () => 10,
        executeSearch: () => {},
        getOpenOrchards: () => [orchard1, orchard2],
        getCloseOrchards: () => [orchard3],
      },
    ];
  },
};

describe("Trigger Period Cycle", () => {
  test("Get Schedule", () => {
    const schedule = triggerPeriodCycle(mockNetworkProvider);
    expect(schedule).toMatchSnapshot();
  });
});
