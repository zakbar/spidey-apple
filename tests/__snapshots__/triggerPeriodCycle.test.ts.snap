// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trigger Period Cycle Get Schedule 1`] = `
{
  "orchardStatuses": [
    {
      "isOpen": true,
      "name": "1",
    },
    {
      "isOpen": true,
      "name": "2",
    },
    {
      "isOpen": false,
      "name": "3",
    },
  ],
}
`;
